lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -120,12 +124,15 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
                 config.getSseKmsKeyId(),
                 Optional.empty(),
                 config.getCannedAcl());
+
+        this.uploadExecutor = newCachedThreadPool(daemonThreadsNamed("s3-upload-%s"));
```

Comment:
Since there are no constructor provided args, you can inline this into the field declaration.